### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by wdwatkins**

### CODE REVIEW

**Comment by wdwatkins** [comment]
Moved this PR over to the USGS-R main branch.  This incorporates @aappling-usgs' s comments from the original PR

**Comment by wdwatkins** [comment]
This should be in good shape now, I have rerun all of it except letting the full 6 initializations complete.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#some of these may still be checked in; also using this
+#to ignore files when zipping repo for data release
+*.ipynb
+run_lstm_check_results.ipynb
```
Comment (aappling-usgs) [inline .gitignore:4]: Is this supposed to have a `!` in front of it? I see this file committed in this PR

File: README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+- train_filter: Apply this filter to forcing data to filter out training and test data
+- test_filter: apply this filter to get *only* the test data. Test/train time period split is handled internally in the modeling code when running, use this to extract sites for a particular test set when checking performance.
+- hidden_sizes:  The sizes of hidden layers of the neural network.  Set these accordingly in StreamTemp-Integ.py
+- batch_size: Size of training batch.  Also set in StreamTemp-Integ.py
+
+To filter the sites and forcing data based on the first experiment in experiments.csv, and write the input files:
+```
+os.makedirs('input/forcing', exist_ok = True)
+os.makedirs('input/attr', exist_ok = True)
+attr_filtered = attr.query('dag >= 60')
```
Comment (aappling-usgs) [inline README.md]: The first experiment in experiments.csv is 
image
which would have `attr.query('dag >= 99')`, wouldn't it?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+### Explanation of different experiments
+
+At this point, you can decide which experiment to replicate. `experiments.csv` in the data release describes the various experiments run in this data release. These columns are most important:
+
+- train_filter: Apply this filter to forcing data to filter out training and test data
+- test_filter: apply this filter to get *only* the test data. Test/train time period split is handled internally in the modeling code when running, use this to extract sites for a particular test set when checking performance.
+- hidden_sizes:  The sizes of hidden layers of the neural network.  Set these accordingly in StreamTemp-Integ.py
+- batch_size: Size of training batch.  Also set in StreamTemp-Integ.py
+
+To filter the sites and forcing data based on the first experiment in experiments.csv, and write the input files:
```
Comment (aappling-usgs) [inline README.md]:
```suggestion
Filter the sites and forcing data based on the `train_filter` field in the row of interest in experiments.csv, then write the input files. For example, for the first row in experiments.csv, `train_filter` is `dag >= 99`, so you can use that filter to create `attr_filtered`. Here we also name the output files ('forcing_99.feather' and 'attr_99.feather') to reflect that filter.
```

File: StreamTemp-Integ.py [hunk 4]
```diff
@@ -98,11 +98,11 @@
         # define training options
         optTrain = default.update(default.optTrainCamels, miniBatch=[BATCH_SIZE, RHO], nEpoch=EPOCH, saveEpoch=saveEPOCH, seed=seed)
         # define output folder for model results
-        exp_name = 'TempDemo'
-        exp_disp = 'FirstRun'
+        #exap_name = 'output'
```
Comment (aappling-usgs) [inline StreamTemp-Integ.py]: `exap`? and commented out? can you just delete these two lines?

File: .gitignore (hunk 1 above)
Comment (wdwatkins) [reply .gitignore:4]: The .gitignore entry is just to ignore it in `copy_zip_repo` so it stays out of the data release

File: README.md (hunk 2 above)
Comment (wdwatkins) [reply README.md]: Right, I had run this one to test another example.  Not being able to run code directly from the readme has led to several copy-paste misses like this.  Maybe it would have been simpler to write the readme in a Jupyter notebook and then export it to a document format 🤔

Comment (aappling-usgs) [reply README.md]: I use Jupyter Lab and an R console within Jupyter Lab, which allows me to copy-paste from the README (or other files) into the console interactively. It's not quite RStudio but works well enough for light development.
